plat/xilinx/zynqmp/pm_service/pm_api_clock.c

```diff
@@ -0,0 +1,3171 @@
+/*
+ * Copyright (c) 2018, ARM Limited and Contributors. All rights reserved.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+/*
+ * ZynqMP system level PM-API functions for clock control.
+ */
+
+#include <arch_helpers.h>
+#include <mmio.h>
+#include <platform.h>
+#include <string.h>
+#include "pm_api_clock.h"
+#include "pm_api_sys.h"
+#include "pm_client.h"
+#include "pm_common.h"
+#include "pm_ipi.h"
+
+#define CLK_NODE_MAX			6
+
+#define CLK_PARENTS_ID_LEN		16
+#define CLK_TOPOLOGY_NODE_OFFSET	16
+#define CLK_TOPOLOGY_PAYLOAD_LEN	12
+#define CLK_PARENTS_PAYLOAD_LEN		12
+#define CLK_INIT_ENABLE_SHIFT		1
+#define CLK_TYPE_SHIFT			2
+#define CLK_CLKFLAGS_SHIFT		8
+#define CLK_TYPEFLAGS_SHIFT		24
+
+#define CLK_EXTERNAL_PARENT	(PARENT_CLK_EXTERNAL << CLK_PARENTS_ID_LEN)
+
+#define NA_MULT					0
+#define NA_DIV					0
+#define NA_SHIFT				0
+#define NA_WIDTH				0
+#define NA_CLK_FLAGS				0
+#define NA_TYPE_FLAGS				0
+
+/* PLL nodes related definitions */
+#define PLL_PRESRC_MUX_SHIFT			20
+#define PLL_PRESRC_MUX_WIDTH			3
+#define PLL_POSTSRC_MUX_SHIFT			24
+#define PLL_POSTSRC_MUX_WIDTH			3
+#define PLL_DIV2_MUX_SHIFT			16
+#define PLL_DIV2_MUX_WIDTH			1
+#define PLL_BYPASS_MUX_SHIFT			3
+#define PLL_BYPASS_MUX_WIDTH			1
+
+/* Peripheral nodes related definitions */
+/* Peripheral Clocks */
+#define PERIPH_MUX_SHIFT			0
+#define PERIPH_MUX_WIDTH			3
+#define PERIPH_DIV1_SHIFT			8
+#define PERIPH_DIV1_WIDTH			6
+#define PERIPH_DIV2_SHIFT			16
+#define PERIPH_DIV2_WIDTH			6
+#define PERIPH_GATE_SHIFT			24
+#define PERIPH_GATE_WIDTH			1
+
+#define USB_GATE_SHIFT				25
+
+/* External clock related definitions */
+
+#define EXT_CLK_MIO_DATA(mio)				\
+	[EXT_CLK_INDEX(EXT_CLK_MIO##mio)] = {		\
+		.name = "mio_clk_"#mio,			\
+	}
+
+#define EXT_CLK_INDEX(n)	(n - CLK_MAX_OUTPUT_CLK)
+
+/* Clock control related definitions */
+#define BIT_MASK(x, y) (((1U << (y)) - 1) << (x))
```

Comment:
can use BIT(y)

Reply:
Hi @sivadur ,
Above macro provides mask at offset x with y width so can not be replaced with BIT(y)